internal/tsc_wrapped/compiler_host.ts

```diff
@@ -343,7 +343,10 @@ export class CompilerHost implements ts.CompilerHost, tsickle.TsickleHost {
    */
   fileExists(filePath: string): boolean {
     // Allow moduleResolution=node to behave normally.
-    if (this.allowNonHermeticReads && isNonHermeticInput(filePath) &&
+    // TODO(alexeagle): the indexOf(node_modules) is a hack, find a better
+    // way to identify these undeclared inputs.
+    if (this.allowNonHermeticReads &&
+        filePath.split(path.sep).indexOf('node_modules') !== -1 &&
```

Comment:
Why is this needed? If everything is an input, why this?

The last time we did an indiscriminate `indexOf(node_modules)` kind of hack we caused a massive regression in google3, maybe it's better to avoid that?

Reply:
What's happening here is I inlined the method which was removed in file_cache. It's not newly introduced.

I agree it would be better to strike any reads from non-declared inputs. I tried removing this condition, and have some typescript module resolution failures which are quite challenging. I'll keep hacking on them.

If I can't figure it out, I'd still like to merge this as it is strictly better performance than what we have today.

Reply:
okay, I found the difference, we were passing the `files[]` array from the tsconfig to satisfy fileExists, but we should ask the cache instead.